src/main/java/nl/knaw/huc/resussun/dataextension/DataExtensionClient.java

```diff
@@ -30,40 +31,68 @@ public DataExtensionClient() {
   public JsonNode createExtensionResponse(ApiData api, DataExtensionRequest extensionRequest)
       throws TimbuctooException {
     final Timbuctoo timbuctoo = api.getTimbuctoo();
-    List<String> propertyIds = extensionRequest.getProperties().stream().map(Property::getId).collect(
-        Collectors.toList());
-    Map<String, String> propertyIdNameMap = createPropertyIdNameMap(api, timbuctoo);
+
+    Map<String, CollectionMetadata> dataSetMetadataMap =
+        timbuctoo.executeRequest(createCollectionsMetadataRequest(api.getDataSourceId()), collectionIdAsKey());
+
     Map<String, Map<String, List<? extends PropertyValue>>> rows = new HashMap<>();
     if (!extensionRequest.getEntityIds().isEmpty()) {
-
+      List<String> propertyIds = extensionRequest.getProperties().stream().map(Property::getId).collect(
+          Collectors.toList());
+      List<PropertyMetadata> properties = dataSetMetadataMap.values().stream()
+                                                            .flatMap(col -> col.getProperties().stream())
+                                                            .filter(prop -> propertyIds.contains(prop.getName()))
+                                                            .collect(Collectors.toList());
```

Comment:
No filtering on the collection the requested entity belongs to? Makes it too difficult I guess? Maybe at the least change the collect to a Set instead, to keep duplicates out. For example, `rdfs:label` can occur in multiple collections.